client/src/pages/signin/Signin.js

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from "react"
+import './Signin.scss'
+import { Link } from 'react-router-dom'
+
+import TextField from '@material-ui/core/TextField';
+import Snackbar from '@material-ui/core/Snackbar';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+
+class signin extends Component{
+
+    constructor(){
+        super()
+
+        this.state={
+            login: {
+                email: '',
+                password: ''
+            },
+            passLength: 0,
+            snackbaropen: false,
+            snackbarmsg: ''
+        }
+    }
+
+    //handles the snackbar warning message
+    snackbarClose = (event) => {
+        this.setState({
+            snackbaropen: false
+        })
+    }
+
+    //handle updating the state with the values of the input fields
+    updateText(event){
+        let updatedText = Object.assign({}, this.state.login)
+        updatedText[event.target.id] = event.target.value
+        
+        let updatedLength
+        event.target.id === 'password' ? updatedLength = event.target.value.length : updatedLength = 0
```

Comment:
Are you using a linter (like prettier.io) - let's invest in that early, since it'll clean up the code and add semicolons and better coding style

Reply:
Enabled linting with prettier with the default config settings and reformatted the code. Thank you for the suggestion.